Centralize semantic tokens for Airflow UI

odaneau-astro opened the pull request:
# Semantic Tokens Design Plan

## Goal

Establish a well-structured semantic tokens hierarchy that provides clear separation between primitive design tokens, system-level semantic tokens, and component-specific tokens. This will improve maintainability, consistency, and developer experience when working with the Airflow UI theme system.

## Current State

The current theme structure mixes reference tokens and semantic tokens without a clear hierarchy. Component-specific tokens are scattered throughout the codebase, making it difficult to maintain consistency and update the design system.

## Proposed Solution: 3-Layer Token Hierarchy

### Layer 1: Reference Tokens

**Definition**: All available Chakra UI "primitive" tokens. These tokens make up all of the style options available in Chakra UI's design system. The tokens point to a static value.

**Example**: `tokens.colors.blue.50` is defined as `"oklch(0.97 0.014 254.604)"`

**Chakra UI Token Types**:

- [Colors](https://www.chakra-ui.com/docs/theming/tokens#colors)
- [Gradients](https://www.chakra-ui.com/docs/theming/tokens#gradients)
- [Sizes](https://www.chakra-ui.com/docs/theming/tokens#sizes)
- [Spacing](https://www.chakra-ui.com/docs/theming/tokens#spacing)
- [Fonts](https://www.chakra-ui.com/docs/theming/tokens#fonts)
- [Font Sizes](https://www.chakra-ui.com/docs/theming/tokens#font-sizes)
- [Font Weights](https://www.chakra-ui.com/docs/theming/tokens#font-weights)
- [Letter Spacings](https://www.chakra-ui.com/docs/theming/tokens#letter-spacings)
- [Line Heights](https://www.chakra-ui.com/docs/theming/tokens#line-heights)
- [Radii](https://www.chakra-ui.com/docs/theming/tokens#radii)
- [Borders](https://www.chakra-ui.com/docs/theming/tokens#borders)
- [Border Widths](https://www.chakra-ui.com/docs/theming/tokens#border-widths)
- [Shadows](https://www.chakra-ui.com/docs/theming/tokens#shadows)
- [Easings](https://www.chakra-ui.com/docs/theming/tokens#easings)
- [Opacity](https://www.chakra-ui.com/docs/theming/tokens#opacity)
- [Z-Index](https://www.chakra-ui.com/docs/theming/tokens#z-index)
- [Assets](https://www.chakra-ui.com/docs/theming/tokens#assets)
- [Durations](https://www.chakra-ui.com/docs/theming/tokens#durations)
- [Animations](https://www.chakra-ui.com/docs/theming/tokens#animations)
- [Aspect Ratios](https://www.chakra-ui.com/docs/theming/tokens#aspect-ratios)

**Purpose**: These are the foundational building blocks. They represent raw design values and should rarely be used directly in components.

### Layer 2: System Tokens

**Definition**: Airflow UI-specific "global" semantic tokens. The tokens point to a reference token and provide semantic meaning.

**Example**: `semanticTokens.colors.info.solid` is defined as `tokens.colors.blue.600`

**Current System Tokens**:
- `brand.*` - Brand colors (solid, fg, bg, muted, subtle, emphasized, contrast, focusRing)
- `warning.*` - Warning/alert colors
- `danger.*` - Danger/error colors
- `error.*` - Error colors
- `info.*` - Informational colors
- `active.*` - Active state colors
- Task state colors: `none.*`, `scheduled.*`, `queued.*`, `running.*`, `success.*`, `restarting.*`, `failed.*`, `skipped.*`, `upstream_failed.*`, `up_for_retry.*`, `up_for_reschedule.*`, `deferred.*`, `removed.*`
- System tokens: `sys.text.*`, `sys.background.*`, `sys.surface.*`, `sys.border.*`, `sys.interaction.*`


**Purpose**: These tokens provide semantic meaning and can be used across multiple components. They abstract away the specific color values and allow for easy theme updates.

### Layer 3: Component Tokens

**Definition**: Single-use component-specific tokens that don't fit into the system-level semantic tokens.

**Purpose**: These tokens are specific to individual components and should be used sparingly. They allow for component-specific customization while maintaining consistency with the overall design system.

## Out of scope: 
1. Chakra UI's styles feature:
- [Text Styles](https://chakra-ui.com/docs/styling/text-styles)
- [Layer Styles](https://chakra-ui.com/docs/styling/layer-styles)
- [Animation Styles](https://chakra-ui.com/docs/styling/animation-styles)
- [Focus Ring](https://chakra-ui.com/docs/styling/focus-ring)

2. [Chakra UI's recipe feature](https://chakra-ui.com/docs/theming/recipes) for component variants.

odaneau-astro committed:
feat: Complete Airflow UI Playground implementation

- Implement comprehensive UI playground showcasing Chakra UI v3 components
- Add sections for color palette, buttons, forms, progress indicators, charts, and graphs
- Update to Chakra UI v3 API patterns and best practices
- Fix all linting and formatting issues
- Add proper accessibility features and semantic HTML
- Implement collapsible sections with consistent styling
- Add task instance states visualization with progress bars
- Include ReactFlow graph components matching DAGs view
- Add comprehensive color matrix display similar to Tailwind CSS
- Remove all custom margins/padding for consistent design
- Add proper TypeScript types and error handling
- Ensure full ESLint and Prettier compliance

Components added:
- Playground.tsx - Main playground page with state management
- ColorPaletteSection.tsx - Color palette matrix display
- ButtonsCodeSection.tsx - Button variations and code examples
- FormsInputsSection.tsx - Form controls and input examples
- ProgressTaskStatesSection.tsx - Progress bars and task states
- ChartsGanttSection.tsx - Chart.js integration with proper theming
- GraphSection.tsx - ReactFlow graph visualization
- BadgesAvatarsSection.tsx - Badge and avatar components
- ModalDialog.tsx - Modal and dialog examples
- TableOfContents.tsx - Navigation with expand/collapse controls
- forms/ - Individual form component examples

Fixes:
- All ESLint errors resolved with proper disable comments where needed
- Prettier formatting applied consistently
- TypeScript compilation errors resolved
- Chakra UI v3 API migration completed
- Accessibility improvements implemented

# Conflicts:
#	airflow-core/src/airflow/ui/src/theme.ts

odaneau-astro committed:
clean up Playground and add missing components

odaneau-astro committed:
Implement a semantic tokens structure in theme.ts and update components to use it.

odaneau-astro committed:
fix TrendCountChart.tsx

odaneau-astro committed:
fix Gantt

odaneau-astro committed:
fix DurationChart

odaneau-astro committed:
fix GridTI

odaneau-astro committed:
fix Graph and AssetGraph

odaneau-astro committed:
remove unused file

odaneau-astro committed:
restore MetricSection

odaneau-astro committed:
fix Stats components

odaneau-astro committed:
revert DeleteVariableButton

odaneau-astro committed:
bulk changes

odaneau-astro committed:
Merge branch 'apache:main' into main

odaneau-astro committed:
remove calls to var(--chakra-colors-*)

odaneau-astro committed:
Fix static check errors

odaneau-astro committed:
Merge branch 'main' into main

rich7420 committed:
fix MyPy type errors in log_reader.py (#56809)

* fix MyPy type errors in log_reader.py

* fix ruff  errors

* remove cast

jscheffl committed:
Fix mypy warnings for SQLA2 migration (#56989)

* Fix mypy warnings for SQLA2 migration

* Revert soucing from TaskInstanceState from common.compat.sdk

KoviAnusha committed:
fix mypy errors in providers/standard/ (#57266)

RoyLee1224 committed:
Fix remaining MyPy type errors in utils/ (#56982)

chiuinggum committed:
Migrate FAB POST /roles to FastAPI (#57199)

* Migrate FAB POST /roles to FastAPI

* Remove unnecessary casting and model

* Validate non-empty role name via Pydantic

* Move auth dep to provider security and fixtureize dependency_overrides

* Add test for roles data model and security provider

KoviAnusha committed:
Fix MyPy type errors in airflow_core models in dag.py,xcom.py for Sqlalchemy 2 migration (#57323)

KoviAnusha committed:
Fix MyPy type errors in /api_fastapi/core_api/routes/public/ for Sqlalchemy 2 migration (#57317)

KoviAnusha committed:
Fix MyPy type errors in /api_fastapi/core_api/ connections.py,pools.py,variables.py for Sqlalchemy 2 migration (#57272)

KoviAnusha committed:
Fix MyPy type errors in mark_tasks.py for Sqlalchemy 2 migration (#57271)

KoviAnusha committed:
Fix MyPy type errors in serialized_objects.py for Sqlalchemy 2 migration (#57269)

nailo2c committed:
Fix MyPy type errors in test_dag_run.py (#57268)

Dev-iL committed:
SQLA2: fix mypy violations in common/parameters.py (#57248)

ris-tlp committed:
SQLA1 -> SQLA2: Mypy fix execution_api/routes/hitl.py (#57239)

* SQLA1 -> SQLA2: Mypy fix execution_api/routes/hitl.py

* Prek ruff format

ris-tlp committed:
Fab provider SQLA1 -> SQLA2: Mypy fix core_api/routes/public/pools.py (#57235)

ris-tlp committed:
Fab provider SQLA1 -> SQLA2: Fix models/dagbundle MyPy error (#57217)

KoviAnusha committed:
Fix MyPy type errors /api_fastapi/execution_api/,/airflow/jobs/, /airflow/models/ in Sqlalchemy 2 migration (#57277)

rich7420 committed:
fix MyPy type errors in api_fastapi/core_api/routes/public/* for SQLAlchemy 2 compatibility (#56813)

dependabot[bot] committed:
Bump the fab-ui-package-updates group across 1 directory with 4 updates (#57160)

* Bump the fab-ui-package-updates group across 1 directory with 4 updates

Bumps the fab-ui-package-updates group with 4 updates in the /providers/fab/src/airflow/providers/fab/www directory: [@babel/core](https://github.com/babel/babel/tree/HEAD/packages/babel-core), [@babel/eslint-parser](https://github.com/babel/babel/tree/HEAD/eslint/babel-eslint-parser), [@babel/plugin-transform-runtime](https://github.com/babel/babel/tree/HEAD/packages/babel-plugin-transform-runtime) and [@babel/preset-env](https://github.com/babel/babel/tree/HEAD/packages/babel-preset-env).


Updates `@babel/core` from 7.28.4 to 7.28.5
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.28.5/packages/babel-core)

Updates `@babel/eslint-parser` from 7.28.4 to 7.28.5
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.28.5/eslint/babel-eslint-parser)

Updates `@babel/plugin-transform-runtime` from 7.28.3 to 7.28.5
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.28.5/packages/babel-plugin-transform-runtime)

Updates `@babel/preset-env` from 7.28.3 to 7.28.5
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.28.5/packages/babel-preset-env)

---
updated-dependencies:
- dependency-name: "@babel/core"
  dependency-version: 7.28.5
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: fab-ui-package-updates
- dependency-name: "@babel/eslint-parser"
  dependency-version: 7.28.5
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: fab-ui-package-updates
- dependency-name: "@babel/plugin-transform-runtime"
  dependency-version: 7.28.5
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: fab-ui-package-updates
- dependency-name: "@babel/preset-env"
  dependency-version: 7.28.5
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: fab-ui-package-updates
...

Signed-off-by: dependabot[bot] <support@github.com>

* Generate hash

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: vincbeck <vincbeck@amazon.com>

vincbeck committed:
Fix mypy error in main (#57351)

Dev-iL committed:
SQLA2: fix mypy violations related to taskinstance (#57279)

Dev-iL committed:
SQLA2: fix mypy violations in `scheduler_job_runner.py` (#57278)

pierrejeambrun committed:
Fix DagRun translation file not found (#57358)

TyrellHaywood committed:
Fix asset name overflow in DAG list view (#57108)

* Fix asset name overflow in DAG list view

- Add maxWidth constraint to HStack containers in AssetSchedule component
- Prevents asset names from overlapping with Latest Run column

* Fix asset name overflow with word-break and line clamping

* Use TruncatedText component instead

---------

Co-authored-by: pierrejeambrun <pierrejbrun@gmail.com>

o-nikolas committed:
Second cut at verifying team/bundle at dag parse time (#57166)

Verify team/bundle/executor at dag parse time

Bundle path is already plumbed down to the dag bag for parsing, also
pass along the bundle name which is used to lookup the bundle config and
find the team name from bundle configuration.
This is used to verify that tasks/dags are using executors that are
available/configured for their team.

potiuk committed:
Move docker to /mnt for the "Publish docs" workflow (#57371)

When we are publishing docs and building SBOMs the regular disk
space on / is not enough to hold all the necessary images. We need
to move it to /mnt to make it works - /mnt has way more space.

wjddn279 committed:
Fix memory leak in Client via SSL context creation (#57334)

related: https://github.com/apache/airflow/issues/56641

When I performed the same memray inspection on the latest version that includes both fixes, the first issue was clearly resolved, but the second issue still persists.
[memray1.html](https://github.com/user-attachments/files/23160226/memray1.html)

When a Client object is created, `ctx = ssl.create_default_context(cafile=ca_file)` continues to be executed repeatedly, which accumulates in memory and causes a memory leak. (It appears to be allocated as a C language object and remains in memory regardless of Python object GC)

This PR uses caching to prevent the SSL context object from being recreated.

Here are the results after running for two hours with this change. The memory usage, which was previously growing to tens of MBs, now stabilizes at approximately 700KB. 
[memray2.html](https://github.com/user-attachments/files/23160405/memray2.html)

gopidesupavan committed:
Fix job label extraction from the workflow jobs response (#57375)

vincbeck committed:
Create list teams API (#57168)

KoviAnusha committed:
Addressed small nits in audit logs documentation (#57319)

* Addressed small nits in audit logs documentation

* Enhance documentation on audit and event logs

Clarified the distinction between audit logs and event logs, including their purposes and functions. Added practical query examples for audit log analysis.

dependabot[bot] committed:
Bump the core-ui-package-updates group across 1 directory with 16 updates (#57362)

* Bump the core-ui-package-updates group across 1 directory with 16 updates

Bumps the core-ui-package-updates group with 16 updates in the /airflow-core/src/airflow/api_fastapi/auth/managers/simple/ui directory:

| Package | From | To |
| --- | --- | --- |
| [@chakra-ui/react](https://github.com/chakra-ui/chakra-ui/tree/HEAD/packages/react) | `3.27.1` | `3.28.0` |
| [@tanstack/react-query](https://github.com/TanStack/query/tree/HEAD/packages/react-query) | `5.90.3` | `5.90.5` |
| [axios](https://github.com/axios/axios) | `1.12.2` | `1.13.0` |
| [@eslint/js](https://github.com/eslint/eslint/tree/HEAD/packages/js) | `9.37.0` | `9.38.0` |
| [@stylistic/eslint-plugin](https://github.com/eslint-stylistic/eslint-stylistic/tree/HEAD/packages/eslint-plugin) | `5.4.0` | `5.5.0` |
| [@vitejs/plugin-react-swc](https://github.com/vitejs/vite-plugin-react/tree/HEAD/packages/plugin-react-swc) | `4.1.0` | `4.2.0` |
| [@vitest/coverage-v8](https://github.com/vitest-dev/vitest/tree/HEAD/packages/coverage-v8) | `3.2.4` | `4.0.4` |
| [eslint](https://github.com/eslint/eslint) | `9.37.0` | `9.38.0` |
| [eslint-plugin-react-hooks](https://github.com/facebook/react/tree/HEAD/packages/eslint-plugin-react-hooks) | `7.0.0` | `7.0.1` |
| [eslint-plugin-react-refresh](https://github.com/ArnaudBarre/eslint-plugin-react-refresh) | `0.4.23` | `0.4.24` |
| [eslint-plugin-unicorn](https://github.com/sindresorhus/eslint-plugin-unicorn) | `61.0.2` | `62.0.0` |
| [happy-dom](https://github.com/capricorn86/happy-dom) | `20.0.2` | `20.0.8` |
| [typescript](https://github.com/microsoft/TypeScript) | `5.8.3` | `5.9.3` |
| [typescript-eslint](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/typescript-eslint) | `8.46.1` | `8.46.2` |
| [vite](https://github.com/vitejs/vite/tree/HEAD/packages/vite) | `7.1.11` | `7.1.12` |
| [vitest](https://github.com/vitest-dev/vitest/tree/HEAD/packages/vitest) | `3.2.4` | `4.0.4` |



Updates `@chakra-ui/react` from 3.27.1 to 3.28.0
- [Release notes](https://github.com/chakra-ui/chakra-ui/releases)
- [Changelog](https://github.com/chakra-ui/chakra-ui/blob/main/packages/react/CHANGELOG.md)
- [Commits](https://github.com/chakra-ui/chakra-ui/commits/@chakra-ui/react@3.28.0/packages/react)

Updates `@tanstack/react-query` from 5.90.3 to 5.90.5
- [Release notes](https://github.com/TanStack/query/releases)
- [Changelog](https://github.com/TanStack/query/blob/main/packages/react-query/CHANGELOG.md)
- [Commits](https://github.com/TanStack/query/commits/@tanstack/react-query@5.90.5/packages/react-query)

Updates `axios` from 1.12.2 to 1.13.0
- [Release notes](https://github.com/axios/axios/releases)
- [Changelog](https://github.com/axios/axios/blob/v1.x/CHANGELOG.md)
- [Commits](https://github.com/axios/axios/compare/v1.12.2...v1.13.0)

Updates `@eslint/js` from 9.37.0 to 9.38.0
- [Release notes](https://github.com/eslint/eslint/releases)
- [Commits](https://github.com/eslint/eslint/commits/v9.38.0/packages/js)

Updates `@stylistic/eslint-plugin` from 5.4.0 to 5.5.0
- [Release notes](https://github.com/eslint-stylistic/eslint-stylistic/releases)
- [Changelog](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint-stylistic/eslint-stylistic/commits/v5.5.0/packages/eslint-plugin)

Updates `@vitejs/plugin-react-swc` from 4.1.0 to 4.2.0
- [Release notes](https://github.com/vitejs/vite-plugin-react/releases)
- [Changelog](https://github.com/vitejs/vite-plugin-react/blob/main/packages/plugin-react-swc/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite-plugin-react/commits/v4.2.0/packages/plugin-react-swc)

Updates `@vitest/coverage-v8` from 3.2.4 to 4.0.4
- [Release notes](https://github.com/vitest-dev/vitest/releases)
- [Commits](https://github.com/vitest-dev/vitest/commits/v4.0.4/packages/coverage-v8)

Updates `eslint` from 9.37.0 to 9.38.0
- [Release notes](https://github.com/eslint/eslint/releases)
- [Commits](https://github.com/eslint/eslint/compare/v9.37.0...v9.38.0)

Updates `eslint-plugin-react-hooks` from 7.0.0 to 7.0.1
- [Release notes](https://github.com/facebook/react/releases)
- [Changelog](https://github.com/facebook/react/blob/main/packages/eslint-plugin-react-hooks/CHANGELOG.md)
- [Commits](https://github.com/facebook/react/commits/HEAD/packages/eslint-plugin-react-hooks)

Updates `eslint-plugin-react-refresh` from 0.4.23 to 0.4.24
- [Release notes](https://github.com/ArnaudBarre/eslint-plugin-react-refresh/releases)
- [Changelog](https://github.com/ArnaudBarre/eslint-plugin-react-refresh/blob/main/CHANGELOG.md)
- [Commits](https://github.com/ArnaudBarre/eslint-plugin-react-refresh/compare/v0.4.23...v0.4.24)

Updates `eslint-plugin-unicorn` from 61.0.2 to 62.0.0
- [Release notes](https://github.com/sindresorhus/eslint-plugin-unicorn/releases)
- [Commits](https://github.com/sindresorhus/eslint-plugin-unicorn/compare/v61.0.2...v62.0.0)

Updates `happy-dom` from 20.0.2 to 20.0.8
- [Release notes](https://github.com/capricorn86/happy-dom/releases)
- [Commits](https://github.com/capricorn86/happy-dom/compare/v20.0.2...v20.0.8)

Updates `typescript` from 5.8.3 to 5.9.3
- [Release notes](https://github.com/microsoft/TypeScript/releases)
- [Changelog](https://github.com/microsoft/TypeScript/blob/main/azure-pipelines.release-publish.yml)
- [Commits](https://github.com/microsoft/TypeScript/compare/v5.8.3...v5.9.3)

Updates `typescript-eslint` from 8.46.1 to 8.46.2
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/typescript-eslint/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v8.46.2/packages/typescript-eslint)

Updates `vite` from 7.1.11 to 7.1.12
- [Release notes](https://github.com/vitejs/vite/releases)
- [Changelog](https://github.com/vitejs/vite/blob/v7.1.12/packages/vite/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite/commits/v7.1.12/packages/vite)

Updates `vitest` from 3.2.4 to 4.0.4
- [Release notes](https://github.com/vitest-dev/vitest/releases)
- [Commits](https://github.com/vitest-dev/vitest/commits/v4.0.4/packages/vitest)

---
updated-dependencies:
- dependency-name: "@chakra-ui/react"
  dependency-version: 3.28.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: core-ui-package-updates
- dependency-name: "@tanstack/react-query"
  dependency-version: 5.90.5
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: core-ui-package-updates
- dependency-name: axios
  dependency-version: 1.13.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: core-ui-package-updates
- dependency-name: "@eslint/js"
  dependency-version: 9.38.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: core-ui-package-updates
- dependency-name: "@stylistic/eslint-plugin"
  dependency-version: 5.5.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: core-ui-package-updates
- dependency-name: "@vitejs/plugin-react-swc"
  dependency-version: 4.2.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: core-ui-package-updates
- dependency-name: "@vitest/coverage-v8"
  dependency-version: 4.0.4
  dependency-type: direct:development
  update-type: version-update:semver-major
  dependency-group: core-ui-package-updates
- dependency-name: eslint
  dependency-version: 9.38.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: core-ui-package-updates
- dependency-name: eslint-plugin-react-hooks
  dependency-version: 7.0.1
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: core-ui-package-updates
- dependency-name: eslint-plugin-react-refresh
  dependency-version: 0.4.24
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: core-ui-package-updates
- dependency-name: eslint-plugin-unicorn
  dependency-version: 62.0.0
  dependency-type: direct:development
  update-type: version-update:semver-major
  dependency-group: core-ui-package-updates
- dependency-name: happy-dom
  dependency-version: 20.0.8
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: core-ui-package-updates
- dependency-name: typescript
  dependency-version: 5.9.3
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: core-ui-package-updates
- dependency-name: typescript-eslint
  dependency-version: 8.46.2
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: core-ui-package-updates
- dependency-name: vite
  dependency-version: 7.1.12
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: core-ui-package-updates
- dependency-name: vitest
  dependency-version: 4.0.4
  dependency-type: direct:development
  update-type: version-update:semver-major
  dependency-group: core-ui-package-updates
...

Signed-off-by: dependabot[bot] <support@github.com>

* Do not upgrade typescript

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: vincbeck <vincbeck@amazon.com>

jscheffl committed:
Adjust authentication token after UI changes in 3.1.1 (#57370)

potiuk committed:
Update release process for source tarballs prepared during release (#57364)

The updated airflow-ctl release process had some small issues related
to source tarball created - which turned out to be just errors in
matching version to tag. When we prepared source tarball we were
using the "x.y.z" tag always - no matter which distribution was
created - this was fine for Airflow but not fine for other
distributions:

* airflow-ctl/x.y.z
* task-sdk/x.y.z
* providers/YYYY-MM-DD

Also names of the source tarballs had extra `apache-' prefix after
we switched to passing actual distribution name as parameter.

This PR fixes the commands used to prepare tarballs:

* proper tag is used for all distributions
* documentation about using those tags is updated
* release documentation for airflow-ctl is updated to have proper
  links
* release documentation for providers is updated to add the new
  way of tagging the providers with `providers/YYYY-MM-DD` tag
  additionally to individual provider's tags (to denote the date
  of provider's wave)

Fixes: #47343

gopidesupavan committed:
Add better logging for jobs label (#57378)

potiuk committed:
Fixes to the release process after airflow-ctl 1.0.0rc2 (#57381)

jscheffl committed:
Add Content-Type to Request Header in Task SDK calls (#57377)

potiuk committed:
Add typer-slim as a common development dependency (#57383)

Seems that old version of fast-api does not have typer as dependency
and when we go to "lowest dependencies" typer is removed even if
fast-api tests need it.

This PR adds typer-slim (which is trimmed-down version of typer
that does not bring a typer cli) as common development dependency.

potiuk committed:
Synchronize boto3 and sqlalchemy dependency lower-binds (#57385)

Generally, we should avoid having dependencies that are not
lower-bounded, because it means that accidentally we can downgrade
to a really low version. Also it's good if our deps have the same
lower-binds everywhere.

olegkachur-e committed:
Add google provider dataflow pipeline streaming system test (#56265)

Add test and update the docs.

Co-authored-by: Oleg Kachur <kachur@google.com>

potiuk committed:
Protect against reproducibility issue with Python client (#57390)

When you are preparing python client with non-default Python
version, the preparation step will fail and let you know how to
reinstall breeze with the default version - because different
versions of Python might produce different generated ast code.

jscheffl committed:
Upgrade prek to most recent version (#57384)

dheerajturaga committed:
Fix Git corruption recovery by moving fetch into retry context (#56913)

* Fix Git corruption recovery by moving fetch into retry context

  Move _fetch_bare_repo call from _ensure_version_in_bare_repo into
  _clone_bare_repo_if_required so that fetch failures (e.g., corrupted packs
  with unresolved deltas) trigger the existing @retry mechanism to clean up
  and re-clone the bare repository automatically.

* Fix tests

kaxil committed:
Update main with Airflow 3.1.1 release details (#57392)

Release: https://pypi.org/project/apache-airflow/3.1.1/

kaxil committed:
Chart: Update Default Airflow image to 3.1.1 (#57393)

https://pypi.org/project/apache-airflow/3.1.1/

Lee-W committed:
fix: make _get_ssl_context_cached a static method (#57401)

r-richmond committed:
Improve clarity on api workers recommendation in docs (#57273)


---------

Co-authored-by: Ash Berlin-Taylor <ash_github@firemirror.com>

xchwan committed:
Enable PT011 rule to airflow-core tests (#57399)

* modify test_dag_serialization

Signed-off-by: Xch1 <qchwan@gmail.com>

* modify test_xcom_arg

Signed-off-by: Xch1 <qchwan@gmail.com>

* modify test_taskinstance

Signed-off-by: Xch1 <qchwan@gmail.com>

* modify test_deadline

Signed-off-by: Xch1 <qchwan@gmail.com>

* modify test_dagrun

Signed-off-by: Xch1 <qchwan@gmail.com>

---------

Signed-off-by: Xch1 <qchwan@gmail.com>

amoghrajesh committed:
Log execution API server URL at task startup (#57343)

Prajwal7842 committed:
[API] Use BulkTaskInstanceBody for patching tis with new state (#57226)

Pyasma committed:
Fix: Handle SQLAlchemy_URI for Apache Impala  (#57153)

* Added SQL alchemy compatible code in impala.py

* Updating Test

* adding two test

* Completed test

* Finish writing test

* Impala-hook

* changed Not in -> !=

amoghrajesh committed:
Respect task retries for signal killed tasks (#55767)

kaxil committed:
Remove callbacks from DAG `default_args` when serializating it (#57397)

eladkal committed:
Relax Kubernetes client version support to <35.0.0 (#57413)

jscheffl committed:
Bugfix/support Subpath w/o Execution API Url (#57372)

* Improve default of execution URL if deployed in subpath

* Improve default of execution URL if deployed in subpath

* Fix unit tests and add coverage for execution URL function

* Fix unit tests and add coverage for execution URL function, ups need mark as test for AF3 only

* Apply suggestion from @amoghrajesh

Co-authored-by: Amogh Desai <amoghrajesh1999@gmail.com>

---------

Co-authored-by: Amogh Desai <amoghrajesh1999@gmail.com>

kacpermuda committed:
Add separate property to decide if task is to be scheduled or marked straight as success (#56039)

The check whether or not to schedule a task is now done within schedule_tis method, I think it would be beneficial to extract that logic into a separate public method.

If task is not being scheduled, the listeners are not called for it - and some listeners would like to know that ahead of time, which would be easier if there was a separate method for that decision. Right now the OL listener was re-creating the logic from the Airflow core, but it became complex and hard to keep up with. I'll have another PR for the OpenLineage provider to check for existence of this new is_schedulable_task method, and call it instead.

RoyLee1224 committed:
fix: Fix mypy assignment errors in HITL models (#57382)

eladkal committed:
Bump snowflake-connector-python>=3.16.0 (#57420)

ephraimbuddy committed:
Remove 'ParamValidationError' handling in apiserver (#57421)

dag.create_dagrun no longer raises ParamValidationError following this
PR: #56923 therefore this usage is no longer needed.

ephraimbuddy committed:
Remove '.git' folder from versions in GitDagBundle (#57069)

* Remove '.git' folder from versions in GitDagBundle

To reduce storage size, this PR removes the git repo in versions when
they are created since the git repo is not necessary on versions.

* Apply suggestion from @ashb

Co-authored-by: Ash Berlin-Taylor <ash_github@firemirror.com>

* fixup! Apply suggestion from @ashb

* Rename arg to prune_dotgit_folder

* Add news fragment item

* Fix typo

* Update comment about prune_dotgit_folder

---------

Co-authored-by: Ash Berlin-Taylor <ash_github@firemirror.com>

eladkal committed:
Bump impyla>=0.22.0 (#57424)

bbovenzi commented on airflow-core/src/airflow/ui/src/pages/Variables/ImportVariablesForm.tsx:
```diff
@@ -154,7 +154,7 @@ const ImportVariablesForm = ({ onClose }: ImportVariablesFormProps) => {
         {isParsing ? (
           <Center mt={2}>
-            <Spinner color="brand.solid" marginRight={2} size="sm" /> Parsing file...
+            <Spinner color="import-variables.spinner.color" marginRight={2} size="sm" /> Parsing file...
```
This feels oddly specific. I feel like `brand.solid` was perfectly fine as a semantic token?

bbovenzi commented on airflow-core/src/airflow/ui/src/theme.ts:
```diff
@@ -392,35 +398,205 @@ export const customConfig = defineConfig({
+            failed: { value: "{colors.red.500}" },
+            running: { value: "{colors.cyan.500}" },
+            queued: { value: "{colors.yellow.500}" },
+            skipped: { value: "{colors.gray.400}" },
```
I think skipped is supposed to be pink, no?

bbovenzi reviewed (commented):
Big fan of moving all our colors to semantic tokens.

I think a number of these colors are too specific like `calendar.hour-label.color` and `test-connection.icon.connected` and we should try reuse more values. If we have a one-off difference, we should see if we can adjust that HTML element to fit an existing pattern before creating a new color token

choo121600 committed:
Add hover rotation animation to AirflowPin (#57326)

* Add hover rotation animation to AirflowPin

* Add _motionSafe prop

tirkarthi committed:
Fix n+1 queries in ui/dags endpoint and use a compact response model. (#57425)

* Use DAGRunLightResponse to stop loading unused fields in response that trigger n+1 queries to other tables.

* Update frontend type and tests.

* Fix MyPy error.

ashb committed:
Add support for "reconnecting" Supervisor Comms and logs in task processes (#57212)

This change relates to #51422, #54706 and gives us a single function that we
can call from the Python Virtual env script to re-connect the SUPERVISOR_COMMS
and logs socket, so variables and connections are accessible from within the
VEnv task, as long as task-sdk is installed.

This also fixes logs produced by a venv task to not be double "encoded" (i.e.
so that we directly see the actual log, not a stringification of the entire
log line with two timestamps etc.)

There is a matching change to the Venv script to use this function, but it is
separated into a different PR so that we don't merge core/sdk and provider
changes in one PR.

The crux of the change in the venv operator (that I've already tested) is this

```diff
+try:
+    from airflow.sdk.execution_time import task_runner
+except ModuleNotFoundError:
+    pass
+else:
+  reinit_supervisor_comms = getattr(task_runner, "reinit_supervisor_comms")
+  if reinit_supervisor_comms:
+      reinit_supervisor_comms()
```

Co-authored-by: Amogh Desai <amoghrajesh1999@gmail.com>

ashb committed:
Allow virtualenv code to access connections/variables and send logs (#57213)

This change is quite simple, as it simply calls the function added in #57212
if it's available. Up until the point that it is released this code will not
hit the `getattr` will return None and be a no-op.

I didn't use the walrus operator here as _theoretically_ someone could be
using this to run in a Python that doesn't support it.

Fixes #51422, fixes #54706

Co-authored-by: Amogh Desai <amoghrajesh1999@gmail.com>

anshuksi282-ksolves committed:
Feat: Add XCom 'Timestamp' and unify task columns (#57053)

* feat: Add XCom 'Timestamp' and unify task columns

* fix: show task_display_name in XCom table with fallback dash

ksharlandjiev committed:
Add SsmGetCommandInvocationOperator and enhance SSM components (#56936)

Add SsmGetCommandInvocationOperator for retrieving SSM command output

This operator allows users to retrieve the output and execution details
from SSM commands, supporting both single instance and all instances scenarios.

Features:
- Retrieve stdout/stderr from SSM command executions
- Support for specific instance or all instances
- Structured output with execution metadata
- Comprehensive error handling

Includes unit tests, documentation updates, and system test integration.

* Pass AWS connection parameters to SsmRunCommandTrigger in deferrable mode

Fixed deferrable SSM commands failing due to missing region configuration:
  * Fix NoRegionError in SsmRunCommandOperator when using deferrable=True
  * Add region_name, verify, and botocore_config parameters to SsmRunCommandTrigger constructor
  * Update SsmRunCommandOperator to pass AWS connection parameters to trigger
  * Add unit tests for trigger serialization and parameter passing

m1racoli committed:
improve deferrable KPO handling of deleted pods in between polls (#56976)

* refactor: log exception in KubernetesPodTrigger

This makes the error's stacktrace easier to review in the logs and does
not depend on `trigger_reentry` to do so.

* refactor: log poll_interval on trigger start

This makes it easier to know which value for poll_interval is actually
being used.

* fix: handle missing pod when trying to cleanup after trigger reentry

It can happen that the pod doesn't exist anymore when running `trigger_reentry`.

In that case `self.hook.get_pod` will cause a 404 API exception, which
then will end up calling `self._clean` which assumed `self.pod` not to be None.

* fix: don't use structured logging when logging poll interval

This doesn't work in Airflow 2.

eladkal committed:
Remove direct dependency on thrift (#57423)

Nataneljpwd committed:
Fix KubernetesExecutor open slots metric (#55797)

* fixed kubernetes executor open_slots issue

* fixed tests

* formatted file

---------

Co-authored-by: Natanel Rudyuklakir <natanelrudyuklakir@gmail.com>

vincbeck committed:
Add support for Airflow 3 in MWAA operators/sensors/triggers (#57443)

amoghrajesh committed:
Fix potential test failures on Py 3.13 against reset_logging (#57417)

dheerajturaga committed:
Fix text selection jumping in logs pane to match text editor behavior (#57309)

* Fix text selection jumping in logs pane to match text editor behavior

  When selecting text by clicking and dragging in the logs pane, the
  selection would jump unexpectedly when clicking in empty space at
  the end of lines or below the last log line. This was caused by the
  virtualized list rendering with absolutely positioned elements that
  didn't extend to fill available space.

  Changes:
  - Make log line containers full width to enable selection in empty areas
  - Add invisible filler element below last log line to prevent EOF jump
  - Ensure minimum height covers all empty space at bottom of container

  The logs pane now behaves like traditional text editors (e.g. Notepad)
  where users can click anywhere and drag to select multiple lines without
  the selection jumping.

* Fix log view vanishing when text wrap is enabled

  When toggling text wrap on in the logs pane, all log content would
  disappear. This was caused by three issues:

  1. CSS textWrap property was set to pre instead of pre-wrap, which
     preserves whitespace but does not wrap text
  2. Virtualizer was not remeasuring items when wrap toggled, causing
     items to be positioned with cached single-line heights even when
     text became multi-line
  3. Width constraints conflicted - items need to be constrained to 100%
     when wrapped (to force wrapping) but max-content when unwrapped
     (to allow horizontal scrolling)

  Changes:
  - Change textWrap from pre to pre-wrap when wrap is enabled
  - Add useLayoutEffect to force virtualizer remeasurement on wrap toggle
  - Add width=100% to VStack to provide proper width reference
  - Set item width conditionally: 100% when wrapped, max-content when not
  - Keep minWidth=100% always for text selection in empty areas

jscheffl committed:
Feature/extract pre commit for airflow core (#57310)

* Extract prek hooks for Airflow Core

* Implement workaround as proposed by @j178

xchwan committed:
Enable PT011 rule to prvoider tests (#56929)

didier-durand committed:
[Doc] Fixing 404 errors for OpenLineage & Oracle providers (#57469)

dabla committed:
Allow a list of SQL statements in GenericTransfer (#56041)

* refactor: Allow a list of SQL to be passed to the GenericTransfer just like the SQLExecuteQueryOperator to avoid the need to use partial/expand which is way more expensive than looping over the SQL-statements within the operator

* refactor: Reformatted execute method in GenericTransfer

* Update providers/common/sql/src/airflow/providers/common/sql/operators/generic_transfer.py

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

* refactor: Reverted list to str for sql parameter in test GenericTransfer

* refactor: Move check of sql parameter in constructor of GenericTransfer

* refactor: Added unit test for GenericTransfer when multiple SQL statements are defined

* Revert "refactor: Move check of sql parameter in constructor of GenericTransfer"

This reverts commit 881f44713a3fd1da074ecda4187f00af24365f6a.

---------

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

AutomationDev85 committed:
Move triggerer from pod-log-reader-role to pod-launcher-role  (#56872)

* Enable triggerer with pod-launcher-role

* Improve unit tests

---------

Co-authored-by: AutomationDev85 <AutomationDev85>

sunank200 committed:
Revert virtualenv connections/variables access and logging as test are failing

Revert virtualenv connections/variables access and logging as tests are failing in https://github.com/apache/airflow/actions/runs/18896380254. The issue here is that it's a socket initialisation error that occurs when CommsDecoder tries to create a socket from stdin (file descriptor 0) in subprocess environments where stdin is not a socket.

This reverts commit fb8b590 (PR #57213) which allowed virtualenv code to access connections/variables and send logs via supervisor comms reinitialization.

sunank200 committed:
Update Oracle provider URL in description (#57478)

ryanahamilton committed:
Refine the visual design, interaction, and accessibility of the global navigation (#57455)

pierrejeambrun committed:
Add number of queries guard in public backfill list endpoints (#57439)

guan404ming committed:
Add Guan-Ming as committer (#57462)

sunank200 committed:
Fix pre-commit zizmor hook version update to fix CI upgrade check failure. (#57476)

pierrejeambrun committed:
Add number of queries guard in public connection list endpoints (#57442)

kaxil committed:
Add correlation-id support to Execution API for request tracing (#57458)

Task SDK clients send `correlation-id` headers with each request to enable
distributed tracing and debugging. The API server now properly handles
these correlation-ids by:

- Binding `correlation-id` to structured logging context for all logs within a request
- Echoing `correlation-id` back in response headers for end-to-end tracing
- Including `correlation-id` in error responses to aid debugging

This enables users to correlate logs across task execution, API requests,
and worker processes using a single unique identifier per request, making
it easier to debug issues in distributed Airflow deployments.

kaxil committed:
Optimize DAG list query for users with limited access (#57460)

When users have limited DAG access, the DAG list query was inefficiently
grouping all DagRuns in the database before filtering. This caused severe
performance degradation in large deployments where a user might access
only a few DAGs out of hundreds or thousands.

The fix filters both the main DAG query and the DagRun subquery by
accessible dag_ids before performing the expensive GROUP BY operation.

Before (queries all dagruns):

```sql
  SELECT ... FROM dag
  LEFT OUTER JOIN (
    SELECT dag_run.dag_id, max(dag_run.id) AS max_dag_run_id
    FROM dag_run
    GROUP BY dag_run.dag_id
  ) AS mrq ON dag.dag_id = mrq.dag_id
```

After (filters to accessible dags):

```sql
  SELECT ... FROM dag
  LEFT OUTER JOIN (
    SELECT dag_run.dag_id, max(dag_run.id) AS max_dag_run_id
    FROM dag_run
    WHERE dag_run.dag_id IN ('accessible_dag_1', 'accessible_dag_2')
    GROUP BY dag_run.dag_id
  ) AS mrq ON dag.dag_id = mrq.dag_id
  WHERE dag.dag_id IN ('accessible_dag_1', 'accessible_dag_2')
```

Performance impact: In a deployment with 100 DAGs (100 runs each) where
a user has access to only 2 DAGs, this reduces the subquery from grouping
10,000 rows down to 200 rows (50x improvement), and eliminates fetching
98 unnecessary DAG models.

Fixes #57427

KoviAnusha committed:
Fix MyPy type errors in airflow_core models, ti_deps, airflow/api/common/ for Sqlalchemy 2 migration (#57405)

eladkal committed:
Update diagrams>=0.24.4 (#57483)

* Update graphviz>=0.21 and diagrams>=0.24.4

* Update diagrams>=0.24.4

vincbeck committed:
Fix `MwaaTaskCompletedTrigger` (#57490)

choo121600 committed:
chore: update navigation z-index using Chakra token (#57493)

Dennnnny committed:
fix: dag_docs content not scrollable when too long (#57509)

jscheffl committed:
Fix Airflowctl release documentation (#57513)

didier-durand committed:
[Doc] Fixing some typos and spelling errors (#57225)

* Fixing some typos and spelling errors

* adding 'chatbot' to spelling_wordlist.txt

* adding 'chatbots' to spelling_wordlist.txt

* removed 'Chatbots' from spelling_wordlist.txt

* Committing new versions of somes files generated by prek

Zinkue committed:
Fix instance_name in UI docs (#57380)

jscheffl committed:
Enable ruff PLW0120 rule (#57456)

uranusjr committed:
Remove useless updated_state in ti_update_state (#57468)

xchwan committed:
Enable PT011 rule to airflow-core tests (#57463)

AutomationDev85 committed:
Unify Pod Startup Tracking: KubernetesPodTriggerer and KubernetesPodOperator Now Share Common Startup Logic (#56875)

* Move container-related functions from PodManager to a separate file

* Moved unit tests

* Sync and async workflow use the same code to track Pod startup

* Reworked unit tests and pod startup logic

* Add api permission error detection for triggerer

* Fix pytest fixture

* Removed not requried code

---------

Co-authored-by: AutomationDev85 <AutomationDev85>

Lzzz666 committed:
Fix MyPy type errors in providers utils/sql.py (#57448)

* Fix MyPy type errors in providers utils/sql.py

* fix: ignore MyPy arg-type error in select().filter()

* Fix: cast filter clauses to ColumnElement for MyPy compatibility

* Refactor: remove unnecessary cast to ColumnElement in filter clauses

Lzzz666 committed:
Fix MyPy type errors in providers openlineage (#57435)

* Fix MyPy type errors in providers utils/utils.py

* Fix mypy error in utils.py by modify type hint for tasksState

pierrejeambrun committed:
Add number of queries guard in public dag tags list endpoints (#57499)

pierrejeambrun committed:
Add query guard in public asset list endpoints (#57438)

kacpermuda committed:
nit: Use new taskinstance method to determine if task will emit OL event (#57446)

pierrejeambrun committed:
Add number of queries guard in public dag warnings list endpoint (#57503)

amoghrajesh committed:
Fix HITL operators failing when using notifiers (#57494)

ephraimbuddy committed:
Eliminate serdag loading added in #56422 (#57537)

It was wrong to load the serdag and not use it. The initial idea was
to use the serdag at line 437 but was omitted. Thinking about it now,
it will be faster to only load serdag when there's a TI associated
with the dag version

pierrejeambrun committed:
Add number of queries guard in public dags list endpoints (#57506)

guan404ming committed:
Add number of queries guard in public xcom entries list endpoints (#55936)

odaneau-astro committed:
Update Playground components to use centralized theme exports

- Replace getReactFlowThemeStyle with useToken + graph.* semantic tokens
  in Playground/graphs/Graph.tsx (new file added to main after branch diverged)
- Replace getComputedCSSVariableValue with resolveTokenValue in
  Playground/charts/Charts.tsx (same rename from main's theme.ts)

Made-with: Cursor
